extensions/git/src/repository.ts

```diff
@@ -844,21 +844,70 @@ export class Repository implements Disposable {
 				}
 			});
 
-			const promises: Promise<void>[] = [];
+			const maxCommandLineLength: number = 30000;
 
 			if (toClean.length > 0) {
-				promises.push(this.repository.clean(toClean));
+				let sliceStart: number = 0;
```

Comment:
These if sections are quite similar in terms of logic. I think that the common slicing part should be extracted into a private method / utility function outside of clean. It could be useful for other repository methods in the future as well.